source/rfc/versioning.rst

```diff
@@ -0,0 +1,141 @@
+
+2. **CentOS**
+
+   *Note: CentOS Release schedule follows that of Red Hat's closely.*
+
+   * `About CentOS <https://wiki.centos.org/About/Product>`_
+
+3. **Ubuntu**
+
+   * `Ubuntu Releases <https://wiki.ubuntu.com/Releases>`_
+
+OpenStack
+^^^^^^^^^
+
+All information about OpenStack releases can be found on `OpenStack Releases website <https://releases.openstack.org>`_.
+
+Red Hat OpenStack Platform
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All information can be found on `Red Hat's Website <https://access.redhat.com/support/policy/updates/openstack/platform>`_.
+
+Handling of upcoming versions
+-----------------------------
+
+Currently, Red Hat/CentOS are using a model where only the newest patch (minor) version is fully supported and support for the olders ones is immediately dropped unless a special license is provided (RHEL EUS). This can lead to unexpected results when the upgrade happens and latest repositories immediately start poining to newer patch-release.
+
+There are two possible kinds of issues here:
+
+1. Unexpected upgrade
+   
+   If the timelines are not tracked correctly and build system uses general repositories from the Internet, builds MAY start failing unexpectedly, leading to immediate and sudden work stops.
+
+2. Lack of support for newer versions
+ 
+   If the products are not tested with upcoming releases ahead of time, we won't know if the update breaks things. 
+
+As a part of the solution, there should be an additional beta-based build. It should run at least weekly in the time near new release of an operating system. It should be non-voting, but logs and reports should be analyzed just as regular builds.
+
+Operating systems
+^^^^^^^^^^^^^^^^^
+
+CentOS and RHEL allow using a *beta* channel for testing with pre-release software (release candidate). All RH-based systems start their beta-repositories empty right after the release and keep pushing new packages before the release of next version.
+
+For CentOS it is available as a CR (`Continuous Release <https://wiki.centos.org/AdditionalResources/Repositories/CR>`_) repository.
+
+For Red Hat repositories with suffix `-beta-rpms` can be used, e.g. `rhel-7-server-beta-rpms`. No additional license required.
+
+Targeting specific releases
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*Note: This section relates to CentOS only.*
+
+To make sure that updates do not happen without operation team's knowledge, current version symlink (7) should be set for currently used point-release manually, e.g. `7 -> 7.5.1804` at the time of writing.
+
+OpenStack
+^^^^^^^^^
+
+OpenStack repositories are currently being used for CentOS only. They might be provided ahead of time if there is a `SIG <https://wiki.centos.org/SpecialInterestGroup>`_ working on packages for that release. The packages are available in `cloud` repositories for CentOS and maintained by `Cloud SIG <https://wiki.centos.org/SpecialInterestGroup/Cloud>`_.
```

Comment:
There are master builds for RDO project available, they can be used to test newer versions in advance. A lot of info here: https://www.rdoproject.org/what/